src/app/frontend/commons/errordialog_service.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Returns the error dialog service
+ *
+ * @param {!md.$dialog} $mdDialog
+ * @return {!ErrorDialog}
+ */
+export default function create($mdDialog) {
+  return new ErrorDialog($mdDialog);
+}
+
+/**
+* Error dialog to display errors which could not be responded properly
+*
+* @final
+**/
+export class ErrorDialog {
```

Comment:
Oh, yes. This needs to be exported if you want to use/reference it somewhere else. That's the idea. Exported things can be used/references in other files. Everything else is private.